contracts/LockedNFT/base/Locked721Base.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.2;
+
+import "./ILocked721Base.sol";
+import "./Locked721AccessControl.sol";
+
+abstract contract Locked721Base is ILocked721Base, Locked721AccessControl {
+    // stores the locked state for each NFT
+    mapping(uint256 => bool) tokenLock;
+
+    // 
+    bool private shouldLockTokensAfterMint = true;
+
+    // this governs the transfer lock functionality
+    // when the token is in a locked state after the transfer, the token cannot be transferred again
+    // until it has been unlocked by tapping the LGT tag and submitting the chip's signature to the API
+    // this effectively turns the NFT into a semi soul bound NFT and prevents the NFT from being traded
+    // without transferring the physical item to the new owner as well
+    bool private shouldPreventTransferWhenLocked = true;
+
+    constructor() {
+      // contract deployer is the admin by default
+      _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+      // contract deployer can be granted some initial roles to start with
+      _setupRole(API_DELEGATE_ROLE, msg.sender);
+    }
+
+    function setShouldPreventTransferWhenLocked(bool lock) external onlyAdmin {
+      shouldPreventTransferWhenLocked = lock;
+    }
+
+    function getShouldPreventTransferWhenLocked() external view returns (bool) {
+      return shouldPreventTransferWhenLocked;
+    }
+
+    // TOKENLOCK FUNCTIONS
+    function _setTokenLock(uint256 tokenId, bool locked) internal {
+      tokenLock[tokenId] = locked;
+    }
+
+    function setTokenLock(uint256 tokenId, bool locked) override external onlyApiDelegate {
+      _setTokenLock(tokenId, locked);
+    }
+
+    // READ TOKENLOCK STATE FUNCTIONS
+    function _getTokenLock(uint256 tokenId) internal virtual view returns (bool) {
+      return tokenLock[tokenId];
+    }
+
+    function getTokenLock(uint256 tokenId) override external view returns (bool) {
+      return _getTokenLock(tokenId);
+    }
+
+    function setShouldLockTokensAfterMint(bool _shouldLockTokensAfterMint) external onlyAdmin {
+      shouldLockTokensAfterMint = _shouldLockTokensAfterMint;
+    }
+
+    function getShouldLockTokensAfterMint() external view returns (bool) {
+      return shouldLockTokensAfterMint;
+    }
+
+    // if a token is locked, then this function returns true, 
+    // unless the token is being minted,
+    // or the message sender has an API DELEGATE role and is claiming the token,
+    // or the token is being burned 
+    function _shouldPreventTokenTransfer (address from, address to, uint256 startTokenId, uint256 batchSize) internal virtual view returns (bool) {
+      // do not prevent token transfer if:
+      if (
+        !shouldPreventTransferWhenLocked || // flag for preventing token transfers
+        from == address(0) || // token is being minted
+        (bytes4(keccak256("claim(address,uint256)")) == msg.sig) || // this transfer is a call to claim
```

Comment:
i changed the logic here to make it more understandable... also there was an edge case that would allow `API_DELEGATE` to transfer (not claim) even when the token was locked... i know that's not an issue in terms of how we use the application, but seemed like a hole we should patch up